src/types/check_config.rs

```diff
@@ -256,4 +274,232 @@ mod tests {
+            Tick::from_time(DateTime::from_timestamp(60 * 25, 0).unwrap()),
+            "us_east".to_string()
+        ));
+        assert!(!multi_region_config_large_interval.should_run(
+            Tick::from_time(DateTime::from_timestamp(60 * 50, 0).unwrap()),
+            "us_west".to_string()
+        ));
+        assert!(multi_region_config_large_interval.should_run(
+            Tick::from_time(DateTime::from_timestamp(60 * 50, 0).unwrap()),
+            "eu_west".to_string()
+        ));
+        assert!(multi_region_config_large_interval.should_run(
+            Tick::from_time(DateTime::from_timestamp(60 * 71, 0).unwrap()),
+            "us_west".to_string()
+        ));
+
+        // Try more complicated times
+        // ((60 * 60) * 13423) Is just selecting a random large hour, then we add minutes to it
+        assert!(multi_region_config_large_interval.should_run(
+            Tick::from_time(DateTime::from_timestamp(((60 * 60) * 13423) + (60 * 15), 0).unwrap()),
+            "us_west".to_string()
+        ));
+        assert!(!multi_region_config_large_interval.should_run(
+            Tick::from_time(DateTime::from_timestamp(((60 * 60) * 13423) + (60 * 39), 0).unwrap()),
+            "us_west".to_string()
+        ));
+        assert!(multi_region_config_large_interval.should_run(
+            Tick::from_time(DateTime::from_timestamp(((60 * 60) * 13423) + (60 * 39), 0).unwrap()),
+            "us_east".to_string()
+        ));
+        assert!(!multi_region_config_large_interval.should_run(
+            Tick::from_time(DateTime::from_timestamp(((60 * 60) * 13423) + (60 * 40), 0).unwrap()),
+            "us_west".to_string()
+        ));
+        assert!(multi_region_config_large_interval.should_run(
+            Tick::from_time(DateTime::from_timestamp(((60 * 60) * 13423) + (60 * 40), 0).unwrap()),
+            "eu_west".to_string()
+        ));
+        assert!(!multi_region_config_large_interval.should_run(
+            Tick::from_time(DateTime::from_timestamp(((60 * 60) * 13423) + (60 * 59), 0).unwrap()),
+            "us_west".to_string()
+        ));
+        assert!(multi_region_config_large_interval.should_run(
+            Tick::from_time(DateTime::from_timestamp(((60 * 60) * 13423) + (60 * 59), 0).unwrap()),
+            "eu_west".to_string()
+        ));
+        assert!(multi_region_config_large_interval.should_run(
+            Tick::from_time(DateTime::from_timestamp(((60 * 60) * 13423) + (60 * 60), 0).unwrap()),
+            "us_west".to_string()
+        ));
+        assert!(!multi_region_config_large_interval.should_run(
+            Tick::from_time(DateTime::from_timestamp(((60 * 60) * 13423) + (60 * 60), 0).unwrap()),
+            "us_east".to_string()
+        ));
+        assert!(!multi_region_config_large_interval.should_run(
+            Tick::from_time(DateTime::from_timestamp(((60 * 60) * 13423) + (60 * 60), 0).unwrap()),
+            "eu_west".to_string()
+        ));
```

Comment:
nit, I might turn this computation into a little helper function or something

```rs
fn test_time(offset: u32) -> DateTime {
  return DateTime::from_timestamp(((60 * 60) * 13423) + (60 * offset), 0).unwrap()
}
```

idk, might make it a bit easier to see whats diff between each call here